File: backend/backend/Areas/Main/Models/CompanyNotes.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Areas.Main.Models;

public class CompanyNotes
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }
    public int CompanyId { get; set; }
    public Company Company { get; set; }
    public int NoteId { get; set; }
    public Note Note { get; set; }
}

File: frontend/lib/models/company_notes.dart
import 'package:frontend/models/company.dart';
import 'package:frontend/models/note.dart';

class CompanyNotes {
  int? id;
  Company? company;
  Note? note;
  int? companyId;
  int? noteId;

  CompanyNotes({
    this.id,
    this.company,
    this.note,
    this.companyId,
    this.noteId
  });

  factory CompanyNotes.fromJson(Map<String, dynamic> json) {
    return CompanyNotes(
      id: json['id'],
      companyId: json['companyId'],
      company: json['company'] != null ? Company.fromJson(json['company']) : null,
      noteId: json['noteId'],
      note: json['note'] != null ? Note.fromJson(json['note']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'task': company,
      'note': note,
      'noteId': noteId,
      'taskId': companyId
    };
  }
}